### ISSUE TITLE
:mod: markup does not work in apidoc generated titles

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1178]
I don't know if it is for my sphinx installation only (not really mine actually) but the generated `:mod:` markup in packages and modules titles are not working (i.e. not recognized at build).

With the following correction to apidoc.py, it works:

```
#!python

# from line 98: comment are replaced lines
        if is_package:
            #heading = ':mod:`&#37;s` Package' &#37; package
            heading = ':mod:`&#37;s` Package' &#37; makename(master_package,subroot)
        else:
            #heading = ':mod:`&#37;s` Module' &#37; py_file
            heading = ':mod:`&#37;s` Module' &#37; makename(master_package,subroot,py_file)
```

I also replaced the function `makename` to allow more than 2 inputs:

```
#!python

def makename(*pkg_and_mod_names): #package, module):
    """Join package, subpackage (...) and module names with a dot."""
    return '.'.join(filter(len,pkg_and_mod_names))
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1178
- Originally reported by: [diener](https://bitbucket.org/diener)
- Originally created at: 2013-05-30T16:26:13.569

**Comment by shimizukawa** [comment]
_From Robert Lehmann on 2013-06-05 22:34:07+00:00_

I did not look into the failure in detail, just two short general remarks:  `filter(None, ...)` seems to be more idiomatic in what you want to achieve.  And submitting patches/pull requests makes it much easier to review your changes for us.  If nobody beats me to it I'll look into the bug sometime soon.

**Comment by shimizukawa** [comment]
_From [diener](https://bitbucket.org/diener) on 2013-06-06 08:40:13+00:00_

Thanks. I did not do a pull request because I am just starting using sphinx and don't know how it works for now. After the 19th June, I will have some time to look into it, if I can help.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2013-09-17 13:28:39+00:00_

I've solved this a bit differently in c84818a785c9.
